### PR TITLE
[feat] Dist Loader for Eval

### PR DESCRIPTION
**Pull request by TongLi3701**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs
- [x] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

* Replaced manual data loading with DistributedDataLoader.
* Fixed bug for using shardformer TP.
* Optimized pipeline. 

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: applications/ColossalEval/colossal_eval/utils/conversation.py [hunk 1]
```diff
@@ -123,9 +123,7 @@ def dict(self):
         }
 
 
-def get_few_shot_prefix(
-    conv: Conversation, few_shot_data: List[str], tokenizer: Optional[AutoTokenizer], language: str, max_tokens: int
-) -> str:
+def get_few_shot_prefix(few_shot_data: List[str], tokenizer: Optional[AutoTokenizer], max_tokens: int) -> str:
```
Comment (Camille7777) [inline applications/ColossalEval/colossal_eval/utils/conversation.py:126]: Remove the `conv` in Args section.

File: applications/ColossalEval/examples/dataset_evaluation/inference.py [hunk 2]
```diff
@@ -35,6 +38,8 @@ def rm_and_merge(
     """
 
     for model_name in model_names:
+        dataset_cat_num_mapping = utils.jload(os.path.join(save_path, model_name, "dataset_cat_num_mapping.json"))
```
Comment (Camille7777) [inline applications/ColossalEval/examples/dataset_evaluation/inference.py:41]: What is this "dataset_cat_num_mapping.json" file?

**Review by Camille7777 — APPROVED** [review]
Some comments added.
Overall LGTM :)

File: applications/ColossalEval/examples/dataset_evaluation/inference.py [hunk 3]
```diff
@@ -192,35 +207,43 @@ def main(args):
             raise ValueError(f"Model class {model_parameter['model_class']} is not a subclass of BaseModel.")
 
         for dataset_name, split_data in inference_data.items():
-            start = 0
+            cat_num_mapping = {}
```
[the remark this answers is not in the document]
Comment (TongLi3701) [reply applications/ColossalEval/examples/dataset_evaluation/inference.py]: This is used to avoid duplicate data in data loader.

**Review by ver217 — APPROVED** [review]

**Pull request merged**